dep0/lib/03_typecheck/src/private/type_assign.hpp

```diff
@@ -0,0 +1,47 @@
+#pragma once
+
+#include "dep0/typecheck/context.hpp"
+#include "dep0/typecheck/error.hpp"
+
+#include "dep0/parser/ast.hpp"
+
+namespace dep0::typecheck {
+
+/**
+ * Assign a type to the given expression in the given context.
+ * 
+ * Currently, not all expressions can be assigned a type;
+ * in particular numerical expressions and initializer lists;
+ * because their type is either context-dependent or bound by a target type.
```

Comment:
Context dependent? I have the context... I must mean something else, so should improve the comment.

Reply:
I see below... What I meant is context-sensitive, meaning that the behaviour of this function might differ depending on what context is passed in. Should try rephrase the comment then